File: Challenges/5/solution.py
ns = [6]

def look_and_say(n):
    x = 1;
    nthTerm = '0'
    print('START OF X LOOP')
    while x <= n:
        currentNthTerm = nthTerm
        nthTerm = ''
        currentDigit = currentNthTerm[0]
        count = 1
        i = 1

        while i <= len(currentNthTerm):
            if x == 1:
                nthTerm = '1'
            elif i == len(currentNthTerm):
                if count != 0:
                    nthTerm = nthTerm + str(count) + currentDigit
                else:
                    nthTerm = nthTerm + currentDigit
            else:
                if currentDigit == currentNthTerm[i]:
                    count += 1
                else:
                    nthTerm = nthTerm + str(count) + currentDigit
                    currentDigit = currentNthTerm[i]
                    count = 1
            i += 1
        
        print(x, nthTerm)
        x += 1
        

for n in ns:
    look_and_say(n)     
    



        
        

File: Challenges/5/solution.java
/**
 * solution
 * 1315
 */
public class solution {

    public static void lookAndSay(int n) {
        int x = 1;
        String nthTerm = "00";

        System.err.println("START OF X LOOP");
        while ( x <= n) {
            String currentNthTerm = nthTerm;
            nthTerm = "";
            char currentDigit = currentNthTerm.charAt(0);
            int count = 1;
            int i = 1;
            
            while (i <= currentNthTerm.length()) {
                if (x == 1) {
                    nthTerm = "1";
                } else if (i == currentNthTerm.length()) {
                    nthTerm = nthTerm + count + currentDigit;             
                } else {
                    if (currentDigit == currentNthTerm.charAt(i)) {
                        count +=1;
                    } else {
                        nthTerm =  nthTerm + count + currentDigit;
                        currentDigit = currentNthTerm.charAt(i);
                        count = 1;
                    }
                }
                i++;
            }
            System.out.println(x + " " + nthTerm);
            x++;
        }
    }

    public static void main(String[] args) {
        lookAndSay(6);
    }
}